Layout.py

```diff
@@ -163,40 +233,60 @@ def resize_pane(self, group, direction, step):
+                if cols[x2] - cols[x1] > MIN_GAP:
+                    cols[x2] -= step
+            else:
+                lx = 0 if x1 - 1 < 0 else x1 - 1
+                if x1 > 0 and cols[x1] - cols[lx] > MIN_GAP:
+                    cols[x1] -= step
+                if x1 == 0 and cols[x2] - cols[x1] > MIN_GAP:
+                    cols[x2] -= step
         elif direction == 'right':
             if x2 == max_x and x1 == 0:
                 return
-            # greater x
-            gx = 1 if x2 + 1 > max_x else x2 + 1
-            if x2 < max_x and cols[gx] - cols[x2] > MIN_GAP:
-                cols[x2] += step
-            if x2 == max_x and cols[x2] - cols[x1] > MIN_GAP:
-                cols[x1] += step
+            if x1 > 0 and x2 < max_x and not alt:
+                if cols[x2 + 1] - cols[x2] > MIN_GAP:
+                    cols[x2] += step
+            elif x1 > 0 and x2 < max_x:
+                if cols[x2] - cols[x2 - 1] > MIN_GAP:
+                    cols[x1] += step
+            else:
+                # greater x
+                gx = 1 if x2 + 1 > max_x else x2 + 1
+                if x2 < max_x and cols[gx] - cols[x2] > MIN_GAP:
+                    cols[x2] += step
+                if x2 == max_x and cols[x2] - cols[x1] > MIN_GAP:
+                    cols[x1] += step
         elif direction == 'up':
             if y1 == 0 and y2 == max_y:
                 return
-            ly = 0 if y1 - 1 < 0 else y1 - 1
-            if y1 > 0 and rows[y1] - rows[ly] > MIN_GAP:
-                rows[y1] -= step
-            if y1 == 0 and rows[y2] - rows[y1] > MIN_GAP:
-                rows[y2] -= step
+            if y1 > 0 and y2 < max_y and not alt:
+                if rows[y2] - rows[y1] > MIN_GAP:
+                    rows[y2] -= step
+            else:
+                ly = 0 if y1 - 1 < 0 else y1 - 1
+                if y1 > 0 and rows[y1] - rows[ly] > MIN_GAP:
+                    rows[y1] -= step
+                if y1 == 0 and rows[y2] - rows[y1] > MIN_GAP:
+                    rows[y2] -= step
         elif direction == 'down':
             if y2 == max_y and y1 == 0:
                 return
-            gy = 1 if y2 + 1 > max_y else y2 + 1
-            if y2 < max_y and rows[gy] - rows[y2] > MIN_GAP:
-                rows[y2] += step
-            if y2 == max_y and rows[y2] - rows[y1] > MIN_GAP:
-                rows[y1] += step
+            if y1 > 0 and y2 < max_y and not alt:
+                if rows[y2 + 1] - rows[y2] > MIN_GAP:
+                    rows[y2] += step
+            elif y1 > 0 and y2 < max_y:
+                if rows[y2] - rows[y2 - 1] > MIN_GAP:
+                    rows[y1] += step
+            else:
+                gy = 1 if y2 + 1 > max_y else y2 + 1
+                if y2 < max_y and rows[gy] - rows[y2] > MIN_GAP:
+                    rows[y2] += step
+                if y2 == max_y and rows[y2] - rows[y1] > MIN_GAP:
+                    rows[y1] += step
```

Comment:
This is how you can resize frames in the middle with different keybindings and alt=true argument

<img src="http://i.imgur.com/WISmYYy.gif">